hooks/terraform-fmt.sh

```diff
@@ -7,11 +7,43 @@ set -e
 # workaround to allow GitHub Desktop to work, add this (hopefully harmless) setting here.
 export PATH=$PATH:/usr/local/bin
 
+write_changes=true
+FILES=()
+
+parse_arguments() {
+	while (($# > 0)); do
+		# Grab param and value splitting on " " or "=" with parameter expansion
+		local PARAMETER="${1%[ =]*}"
+		local VALUE="${1#*[ =]}"
+		if [[ "$PARAMETER" == "$VALUE" ]]; then VALUE="$2"; fi
+		shift
+		case "$PARAMETER" in
+		--no-autofix)
+			write_changes=false
+			;;
+		-*)
+			echo "Error: Unknown option: $PARAMETER" >&2
+			exit 1
+			;;
+		*)
+			FILES+=("$PARAMETER")
+			;;
+		esac
+	done
+}
+
+parse_arguments "$@"
+
 # Store and return last failure from fmt so this can validate every directory passed before exiting
 FMT_ERROR=0
 
-for file in "$@"; do
-  terraform fmt -diff -check "$file" || FMT_ERROR=$?
+for file in "$FILES"; do
+  file=$(dirname "$file")
```

Comment:
This is required to be able to run the hook outside the repo folder. For example, `pre-commit try-repo /path/to/local/hook terraform-fmt` would invoke `terraform fmt` with files relative to that `/path/to/local/hook`. By providing the entire dirname, we can use the pre-commit tool for local development easily.